vm/src/builtins.rs

```diff
@@ -125,9 +124,11 @@ fn builtin_compile(args: CompileArgs, vm: &VirtualMachine) -> PyResult<PyCodeRef
         Either::B(bytes) => str::from_utf8(&bytes).unwrap().to_string(),
     };
 
-    let mode = get_compile_mode(vm, &args.mode.value)?;
+    let mode = args.mode.as_str().parse().map_err(|_| {
+        vm.new_value_error("compile() mode must be 'exec', 'eval' or 'single'".to_owned())
```

Comment:
I think you can use the display method of the parse error for the error message.